app/apps/handlers.js

```diff
@@ -37,10 +37,12 @@ exports.create = (req, res, next) => {
       } else if (req.body['new-app-datasource'] === 'select') {
         grant_access = created_app.grant_bucket_access(req.body['select-existing-datasource'], 'readonly');
       }
-      grant_access.then(() => {
-        const { url_for } = require('../routes'); // eslint-disable-line global-require
-        res.redirect(url_for('apps.details', { id: created_app.id }));
-      });
+      grant_access
+        .then(() => created_app.grant_user_access(req.user.user_id, 'readwrite', true))
```

Comment:
This needs curly braces - grant_user_access has side-effects.

Reply:
Doh